src/lib/coda_networking/coda_networking.ml

```diff
@@ -523,14 +523,22 @@ module Make (Inputs : Inputs_intf) = struct
     let get_staged_ledger_aux_and_pending_coinbases_at_hash_rpc conn ~version:_
         hash =
       let hash_in_envelope = wrap_rpc_data_in_envelope conn hash in
```

Comment:
Can you write a combinator that abstracts out the common pattern from these functions? We construct an envelope from the rpc parameter and connection object, record `Made_request`, then pass the enveloped argument to some handler function. It'd make the code a little less repetitive.

Reply:
OK